src/integrations/PinterestTag/browser.js

```diff
@@ -229,13 +268,15 @@ export default class PinterestTag {
       pageObject.category = category;
       event = "ViewCategory";
     }
+    this.setLdp(rudderElement.message);
     window.pintrk("track", event, pageObject);
   }
 
   identify() {
     const email = this.analytics.userTraits && this.analytics.userTraits.email;
     if (email) {
-      window.pintrk("set", { em: email });
+      const ldpObject = this.generateLdpObject();
+      window.pintrk('set', { em: email, ...ldpObject });
```

Comment:
What happened if `ldpObject` is undefined or empty?

Reply:
`generateLdpObject()` will always return non empty ldpObject

Reply:
generateLdpObject() will return non empty ldpObject